src/transformers/pipelines/base.py

```diff
@@ -79,7 +79,14 @@ def _pad(items, key, padding_value, padding_side):
             # This is probable image so padding shouldn't be necessary
             # B, C, H, W
             return torch.cat([item[key] for item in items], dim=0)
-        max_length = max(item[key].shape[1] for item in items)
+
+        # No need to perform padding if all lengths are equal
```

Comment:
Necessary to make image classification pipeline test (with `ImageGPTForImageClassification`) pass.